spec/EmailVerificationToken.spec.js

```diff
@@ -3,295 +3,230 @@
 const Auth = require('../lib/Auth');
 const Config = require('../lib/Config');
 const request = require('../lib/request');
+const { resolvingPromise, sleep } = require('../lib/TestUtils');
 const MockEmailAdapterWithOptions = require('./support/MockEmailAdapterWithOptions');
 
-describe('Email Verification Token Expiration: ', () => {
-  it('show the invalid verification link page, if the user clicks on the verify email link after the email verify token expires', done => {
+describe('Email Verification Token Expiration:', () => {
+  it('show the invalid verification link page, if the user clicks on the verify email link after the email verify token expires', async ()  => {
     const user = new Parse.User();
     let sendEmailOptions;
+    const sendPromise = resolvingPromise();
     const emailAdapter = {
       sendVerificationEmail: options => {
         sendEmailOptions = options;
+        sendPromise.resolve();
       },
       sendPasswordResetEmail: () => Promise.resolve(),
       sendMail: () => {},
     };
-    reconfigureServer({
+    await reconfigureServer({
       appName: 'emailVerifyToken',
       verifyUserEmails: true,
       emailAdapter: emailAdapter,
       emailVerifyTokenValidityDuration: 0.5, // 0.5 second
       publicServerURL: 'http://localhost:8378/1',
-    })
-      .then(() => {
-        user.setUsername('testEmailVerifyTokenValidity');
-        user.setPassword('expiringToken');
-        user.set('email', 'user@parse.com');
-        return user.signUp();
-      })
-      .then(() => {
-        // wait for 1 second - simulate user behavior to some extent
-        setTimeout(() => {
-          expect(sendEmailOptions).not.toBeUndefined();
+    });
+    user.setUsername('testEmailVerifyTokenValidity');
+    user.setPassword('expiringToken');
+    user.set('email', 'user@parse.com');
+    await user.signUp();
+    await sendPromise;
+    // wait for 1 second - simulate user behavior to some extent
+    await sleep(1000);
```

Comment:
Can all these sleeps be reduced to maybe 100ms or less? I think it makes sense to introduce a global var or sleep method for this specific purpose like `const emailSendSleepDuration = 100;`, so we can manage them easier.

Reply:
There are only 2 sleep calls and these require to be 1 second per the original comment. 100ms causes them to fail